modyn/config/schema/pipeline.py

```diff
@@ -418,16 +418,35 @@ class DataConfig(BaseModel):
 # ------------------------------------------------------ TRIGGER ----------------------------------------------------- #
 
 
-class TriggerConfig(BaseModel):
-    id: str = Field(description="Type of trigger to be used.")
+class TimeTriggerConfig(BaseModel):
+    id: Literal["TimeTrigger"] = Field("TimeTrigger")
+    every: int = Field(description="The interval length for the trigger specified by an integer", ge=1)
+    unit: Literal["s", "m", "h", "d", "w", "mth", "y"] = Field(description="The unit of the interval length.")
```

Comment:
I'd actually like (in a Pythonic way) to also allow strings like "1m" but I guess that makes Pydantic validation harder. If that is a big problem, we can stick to the updated version.

Reply:
There are a couple of options to stick with the old config (e.g. "every: 1s") while keeping the splitting logic in the pydantic (and hence away from the business logic):
1) composite string field + two (maybe cached) property functions that splits up the composite string
2) adding a model validator in the before mode. (that can change the model inputs before pydantic sees them). As we would need a type change (str -> int) here, this would be rather nasty.

So let's go for 1, I agree that the old config format is more user-friendly :)

Reply:
Thanks for integrating this!